File: src/main/java/io/github/bbortt/buessle/config/DevCorsConfiguration.java
package io.github.bbortt.buessle.config;

import org.springframework.boot.context.config.Profiles;
import org.springframework.context.annotation.Configuration;
import org.springframework.context.annotation.Profile;
import org.springframework.web.servlet.config.annotation.CorsRegistry;
import org.springframework.web.servlet.config.annotation.EnableWebMvc;
import org.springframework.web.servlet.config.annotation.WebMvcConfigurer;

@EnableWebMvc
@Configuration
@Profile({"dev"})
public class DevCorsConfiguration implements WebMvcConfigurer {

  @Override
  public void addCorsMappings(CorsRegistry registry) {
    registry.addMapping("/**");
  }
}

File: src/main/kotlin/io/github/bbortt/buessle/app/config/DevCorsConfiguration.kt
package io.github.bbortt.buessle.app.config

import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration
import org.springframework.context.annotation.Profile
import org.springframework.web.cors.CorsConfiguration
import org.springframework.web.cors.UrlBasedCorsConfigurationSource
import org.springframework.web.filter.CorsFilter
import java.util.*

@Configuration
@Profile("dev")
class DevCorsConfiguration(corsFilter: UrlBasedCorsConfigurationSource = corsConfiguration()) : CorsFilter(corsFilter) {

    companion object {
        @Bean
        fun corsConfiguration(): UrlBasedCorsConfigurationSource {
            val configuration = CorsConfiguration()
            configuration.allowedOrigins = Collections.singletonList("http://localhost:3000")
            configuration.allowedMethods = Arrays.asList("GET", "POST")
            configuration.allowedHeaders = Collections.singletonList("content-type")
            val source = UrlBasedCorsConfigurationSource()
            source.registerCorsConfiguration("/api/**", configuration)
            return source
        }
    }
}
